homeassistant/helpers/service.py

```diff
@@ -113,3 +117,58 @@ def extract_entity_ids(hass, service_call):
         return group.expand_entity_ids(hass, [service_ent_id])
 
     return [ent_id for ent_id in group.expand_entity_ids(hass, service_ent_id)]
+
+
+# pylint: disable=too-few-public-methods
+class CallRateDelayThrottle(object):
```

Comment:
I think that this might be a nice addition to the HA core in the future but in that case it should be async. Let's just keep this for now embedded in the pilight component.